src/handler/sign_tx.c

```diff
@@ -217,46 +217,44 @@ bool sign_tx_with_key() {
     }
 
     cx_ecfp_private_key_t private_key = {0};
-    cx_ecfp_public_key_t public_key = {0};
 
     uint8_t chain_code[32];
 
     derive_private_key(&private_key,
                        chain_code,
                        G_context.bip32_path.path,
                        G_context.bip32_path.length);
-    init_public_key(&private_key, &public_key);
 
     if (G_context.tx_info.sighash_all[0] == '\0') {
         // finish sha256 from data
-        cx_hash(&G_context.tx_info.sha256.header,
-                CX_LAST,
-                G_context.tx_info.sighash_all,
-                0,
-                G_context.tx_info.sighash_all,
-                32);
+        CX_THROW(cx_hash_no_throw(&G_context.tx_info.sha256.header,
+                                  CX_LAST,
+                                  G_context.tx_info.sighash_all,
+                                  0,
+                                  G_context.tx_info.sighash_all,
+                                  32));
         // now get second sha256
         cx_sha256_init(&G_context.tx_info.sha256);
-        cx_hash(&G_context.tx_info.sha256.header,
-                CX_LAST,
-                G_context.tx_info.sighash_all,
-                32,
-                G_context.tx_info.sighash_all,
-                32);
+        CX_THROW(cx_hash_no_throw(&G_context.tx_info.sha256.header,
+                                  CX_LAST,
+                                  G_context.tx_info.sighash_all,
+                                  32,
+                                  G_context.tx_info.sighash_all,
+                                  32));
     }
 
     uint8_t out[256] = {0};
-    size_t sig_size = cx_ecdsa_sign(&private_key,
+    size_t sig_size = 256;
```

Comment:
signature size is not returned by the `_no_throw` variant, it uses the output buffer length pointer to return the information.
So we need to change `sig_size` to be the output length and pass a pointer to it.